In [1]:
import numpy as np 
import pandas as pd 

In [2]:
import json 
from sklearn import preprocessing

In [3]:
import matplotlib.pyplot as plt

In [4]:
import math as m 

In [5]:
%load_ext ipython_memory_usage

Enabling IPython Memory Usage, use %imu_start to begin, %imu_stop to end


In [6]:
%imu_start 

'IPython Memory Usage started'

In [6] used 0.4 MiB RAM in 0.12s (system mean cpu 0%, single max cpu 0%), peaked 0.0 MiB above final usage, current RAM usage now 151.0 MiB


### EDA de la dataFrame 'Athlete_events'

In [7]:
df_athlete=pd.read_csv('data/athlete_events.csv',sep=',')

df_athlete

Name Sex   Age  Height  Weight  \
0                      A Dijiang   M  24.0  180.00    80.0   
1                       A Lamusi   M  23.0  170.00    60.0   
2            Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3           Edgar Lindenau Aabye   M  34.0     NaN     NaN   
4       Christine Jacoba Aaftink   F  21.0  185.00    82.0   
...                          ...  ..   ...     ...     ...   
282985              ZOU Jingyuan   M  23.0    1.58     NaN   
282986          ZUBIMENDI Martin   M  22.0    1.80     NaN   
282987            ZUEV Alexander   M  25.0    1.93     NaN   
282988          ZVEREV Alexander   M  24.0    1.98     NaN   
282989              ZYKOVA Yulia   F  26.0     NaN     NaN   

                              Team  NOC        Games  Year  Season       City  \
0                            China  CHN  1992 Summer  1992  Summer  Barcelona   
1                            China  CHN  2012 Summer  2012  Summer     London   
2                          Denmark  DEN  1920 Summer  1920  Summer  Antwerpen   
3                   Denmark/Sweden  DEN  1900 Summer  1900  Summer      Paris   
4                      Netherlands  NED  1988 Winter  1988  Winter    Calgary   
...                            ...  ...          ...   ...     ...        ...   
282985  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282986                       Spain  ESP   Tokyo 2020  2020  Summer      Tokyo   
282987                         ROC  ROC   Tokyo 2020  2020  Summer      Tokyo   
282988                     Germany  GER   Tokyo 2020  2020  Summer      Tokyo   
282989                         ROC  ROC   Tokyo 2020  2020  Summer      Tokyo   

                      Sport                             Event   Medal  
0                Basketball       Basketball Men's Basketball     NaN  
1                      Judo      Judo Men's Extra-Lightweight     NaN  
2                  Football           Football Men's Football     NaN  
3                Tug-Of-War       Tug-Of-War Men's Tug-Of-War    Gold  
4             Speed Skating  Speed Skating Women's 500 metres     NaN  
...                     ...                               ...     ...  
282985  Artistic Gymnastics               Men's Parallel Bars    Gold  
282986             Football                               Men  Silver  
282987       3x3 Basketball                               Men  Silver  
282988               Tennis                     Men's Singles    Gold  
282989             Shooting       50m Rifle 3 Positions Women  Silver  

[282990 rows x 14 columns]

In [7] used 73.1 MiB RAM in 3.84s (system mean cpu 34%, single max cpu 80%), peaked 0.0 MiB above final usage, current RAM usage now 224.1 MiB


### DataFrame dictionnary

In [8]:
df_dictionnary=pd.read_csv('data/dictionary.csv')

df_dictionnary=df_dictionnary.rename(columns={'GDP per Capita':'PIB/habitant','Code':'NOC'})

df_dictionnary.isna().sum()

Country          0
NOC              0
Population       5
PIB/habitant    25
dtype: int64

In [8] used 3.5 MiB RAM in 1.04s (system mean cpu 27%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 227.6 MiB


### On cherche les athlètes qui n'ont pas de team et ceux des îles TUVALU

In [9]:
df_athlete[df_athlete['NOC']=='UNK']


Name Sex  Age  Height  Weight     Team  NOC        Games  \
61080   Fritz Eccard   M  NaN     NaN     NaN  Unknown  UNK  1912 Summer   
130721     A. Laffen   M  NaN     NaN     NaN  Unknown  UNK  1912 Summer   

        Year  Season       City             Sport  \
61080   1912  Summer  Stockholm  Art Competitions   
130721  1912  Summer  Stockholm  Art Competitions   

                                      Event Medal  
61080   Art Competitions Mixed Architecture   NaN  
130721  Art Competitions Mixed Architecture   NaN

In [9] used 0.8 MiB RAM in 0.14s (system mean cpu 49%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 228.4 MiB


Fritez Eccard a participé dans l'équipe Suédoise et A.Laffen .... aucunes traces je sais pas qui est ce mec 

In [10]:
df_athlete[df_athlete['NOC']=='ROT'].nunique()


Name      10
Sex        2
Age        7
Height     7
Weight     9
Team       1
NOC        1
Games      1
Year       1
Season     1
City       1
Sport      3
Event     12
Medal      0
dtype: int64

In [10] used 0.4 MiB RAM in 0.14s (system mean cpu 34%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 228.8 MiB


12 participations parmi 10 athlètes qui ont participé aux JO ces dernières années sous la team des athlètes réfugiés. 

In [11]:
df_athlete=df_athlete.query(" NOC !='ROC' and NOC != 'UNK' ")

In [11] used 6.0 MiB RAM in 0.16s (system mean cpu 41%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 234.8 MiB


## Base données région 

In [12]:
df_regions=pd.read_csv('data/noc_regions.csv')

df_regions=df_regions.rename(columns={'region':'Country'})
df_regions

NOC      Country                 notes
0    AFG  Afghanistan                   NaN
1    AHO      Curacao  Netherlands Antilles
2    ALB      Albania                   NaN
3    ALG      Algeria                   NaN
4    AND      Andorra                   NaN
..   ...          ...                   ...
225  YEM        Yemen                   NaN
226  YMD        Yemen           South Yemen
227  YUG       Serbia            Yugoslavia
228  ZAM       Zambia                   NaN
229  ZIM     Zimbabwe                   NaN

[230 rows x 3 columns]

In [12] used -4.5 MiB RAM in 0.63s (system mean cpu 33%, single max cpu 100%), peaked 4.5 MiB above final usage, current RAM usage now 230.3 MiB


In [13]:
df_regions.nunique()

NOC        230
Country    206
notes       21
dtype: int64

In [13] used -0.2 MiB RAM in 0.11s (system mean cpu 41%, single max cpu 100%), peaked 0.2 MiB above final usage, current RAM usage now 230.1 MiB


Il y a des régions qui ont le même NOC 

In [14]:
duplicates=[df_regions.duplicated(subset='Country')]

In [14] used -8.2 MiB RAM in 0.11s (system mean cpu 41%, single max cpu 100%), peaked 8.2 MiB above final usage, current RAM usage now 221.9 MiB


In [15]:
df_regions[df_regions.duplicated(subset='Country')]

NOC         Country                   notes
12   AUS       Australia                     NaN
54   CZE  Czech Republic                     NaN
76   GDR         Germany                     NaN
79   GER         Germany                     NaN
81   GRE          Greece                     NaN
88   HKG           China               Hong Kong
126  MAS        Malaysia                     NaN
143  NBO        Malaysia            North Borneo
147  NFL          Canada            Newfoundland
171  RUS          Russia                     NaN
173  SAA         Germany                     NaN
185  SRB          Serbia                     NaN
197  TCH  Czech Republic                     NaN
208  TUV             NaN                  Tuvalu
210  UAR           Syria    United Arab Republic
213  UNK             NaN                 Unknown
214  URS          Russia                     NaN
222  VNM         Vietnam                     NaN
223  WIF        Trinidad  West Indies Federation
225  YEM           Yemen                     NaN
226  YMD           Yemen             South Yemen
227  YUG          Serbia              Yugoslavia
229  ZIM        Zimbabwe                     NaN

In [15] used -0.8 MiB RAM in 0.12s (system mean cpu 46%, single max cpu 100%), peaked 0.8 MiB above final usage, current RAM usage now 221.1 MiB


In [16]:
df_regions.isnull().sum()

NOC          0
Country      3
notes      209
dtype: int64

In [16] used -5.6 MiB RAM in 0.11s (system mean cpu 45%, single max cpu 100%), peaked 5.6 MiB above final usage, current RAM usage now 215.5 MiB


In [17]:
df_regions[df_regions['Country'].isnull()]

NOC Country                 notes
168  ROT     NaN  Refugee Olympic Team
208  TUV     NaN                Tuvalu
213  UNK     NaN               Unknown

In [17] used -19.8 MiB RAM in 0.13s (system mean cpu 49%, single max cpu 100%), peaked 19.8 MiB above final usage, current RAM usage now 195.7 MiB


Il y a  3 des regions qui ne sont pas attribuées, ROT ( refugee olympic team)  , TUVALU ( 9 archipels qui constituent un État ) et UNK qui correspond à Unknown 

### On ajoute les iles TUVALU 

In [18]:
df_regions[df_regions['Country'].isnull()]

NOC Country                 notes
168  ROT     NaN  Refugee Olympic Team
208  TUV     NaN                Tuvalu
213  UNK     NaN               Unknown

In [18] used -6.1 MiB RAM in 0.11s (system mean cpu 40%, single max cpu 100%), peaked 6.1 MiB above final usage, current RAM usage now 189.5 MiB


In [19]:
df_regions=df_regions.query(" NOC != 'ROT' and NOC != 'UNK' ")

df_regions.isnull().sum()

NOC          0
Country      1
notes      209
dtype: int64

In [19] used -6.6 MiB RAM in 0.12s (system mean cpu 40%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 182.9 MiB


In [20]:
df_regions['Country']=df_regions['Country'].fillna('Tuvalu')

In [20] used -12.0 MiB RAM in 0.11s (system mean cpu 11%, single max cpu 50%), peaked 0.0 MiB above final usage, current RAM usage now 170.9 MiB


<ipython-input-20-0f7f308fe64b>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_regions['Country']=df_regions['Country'].fillna('Tuvalu')


In [21]:
df_regions[df_regions['Country']=='Tuvalu']

NOC Country   notes
208  TUV  Tuvalu  Tuvalu

In [21] used 0.1 MiB RAM in 0.11s (system mean cpu 19%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 171.0 MiB


In [22]:
df_regions.isnull().sum()

NOC          0
Country      0
notes      209
dtype: int64

In [22] used -6.4 MiB RAM in 0.12s (system mean cpu 28%, single max cpu 100%), peaked 6.4 MiB above final usage, current RAM usage now 164.7 MiB


# On va merge les dataframes 

## Merge en left 

In [23]:
df_complete=pd.merge(pd.merge(df_athlete,df_regions,on='NOC',how='left'),df_dictionnary,on='NOC',how='left')
#df_complete=df_complete.drop(labels='',axis=1)

df_complete

Name Sex   Age  Height  Weight  \
0                      A Dijiang   M  24.0  180.00    80.0   
1                       A Lamusi   M  23.0  170.00    60.0   
2            Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3           Edgar Lindenau Aabye   M  34.0     NaN     NaN   
4       Christine Jacoba Aaftink   F  21.0  185.00    82.0   
...                          ...  ..   ...     ...     ...   
282622        ZOLOTIC Anastasija   F  19.0    1.80     NaN   
282623              ZOU Jingyuan   M  23.0    1.58     NaN   
282624              ZOU Jingyuan   M  23.0    1.58     NaN   
282625          ZUBIMENDI Martin   M  22.0    1.80     NaN   
282626          ZVEREV Alexander   M  24.0    1.98     NaN   

                              Team  NOC        Games  Year  Season       City  \
0                            China  CHN  1992 Summer  1992  Summer  Barcelona   
1                            China  CHN  2012 Summer  2012  Summer     London   
2                          Denmark  DEN  1920 Summer  1920  Summer  Antwerpen   
3                   Denmark/Sweden  DEN  1900 Summer  1900  Summer      Paris   
4                      Netherlands  NED  1988 Winter  1988  Winter    Calgary   
...                            ...  ...          ...   ...     ...        ...   
282622    United States of America  USA   Tokyo 2020  2020  Summer      Tokyo   
282623  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282624  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282625                       Spain  ESP   Tokyo 2020  2020  Summer      Tokyo   
282626                     Germany  GER   Tokyo 2020  2020  Summer      Tokyo   

                      Sport                             Event   Medal  \
0                Basketball       Basketball Men's Basketball     NaN   
1                      Judo      Judo Men's Extra-Lightweight     NaN   
2                  Football           Football Men's Football     NaN   
3                Tug-Of-War       Tug-Of-War Men's Tug-Of-War    Gold   
4             Speed Skating  Speed Skating Women's 500 metres     NaN   
...                     ...                               ...     ...   
282622            Taekwondo                       Women -57kg    Gold   
282623  Artistic Gymnastics                        Men's Team  Bronze   
282624  Artistic Gymnastics               Men's Parallel Bars    Gold   
282625             Football                               Men  Silver   
282626               Tennis                     Men's Singles    Gold   

          Country_x notes      Country_y    Population  PIB/habitant  
0             China   NaN          China  1.371220e+09   8027.683810  
1             China   NaN          China  1.371220e+09   8027.683810  
2           Denmark   NaN        Denmark  5.676002e+06  51989.293471  
3           Denmark   NaN        Denmark  5.676002e+06  51989.293471  
4       Netherlands   NaN    Netherlands  1.693652e+07  44299.768085  
...             ...   ...            ...           ...           ...  
282622          USA   NaN  United States  3.214188e+08  56115.718426  
282623        China   NaN          China  1.371220e+09   8027.683810  
282624        China   NaN          China  1.371220e+09   8027.683810  
282625        Spain   NaN          Spain  4.641827e+07  25831.582305  
282626      Germany   NaN        Germany  8.141314e+07  41313.313995  

[282627 rows x 19 columns]

In [23] used 71.8 MiB RAM in 0.28s (system mean cpu 24%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 236.5 MiB


In [24]:
df_complete=df_complete.query(" NOC !='ROC' ")

In [24] used -7.0 MiB RAM in 0.16s (system mean cpu 17%, single max cpu 100%), peaked 7.0 MiB above final usage, current RAM usage now 229.5 MiB


In [25]:
#df_complete.info()

In [25] used 0.1 MiB RAM in 0.11s (system mean cpu 15%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 229.6 MiB


282 627 lignes , c'est normal on a enlevé les  réfugiés

In [26]:
df_complete.loc[(df_complete['Medal'].isna(),'Medal')]='Pas_medaille'



In [26] used 0.7 MiB RAM in 0.13s (system mean cpu 18%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 230.3 MiB


## Encoding test 

In [27]:
dictio_score_medaille={'Pas_medaille':0,
                      'Bronze':1,
                      'Silver':2,
                      'Gold':3}

df_complete['Score_medaille']=df_complete['Medal'].map(dictio_score_medaille)

df_complete

Name Sex   Age  Height  Weight  \
0                      A Dijiang   M  24.0  180.00    80.0   
1                       A Lamusi   M  23.0  170.00    60.0   
2            Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3           Edgar Lindenau Aabye   M  34.0     NaN     NaN   
4       Christine Jacoba Aaftink   F  21.0  185.00    82.0   
...                          ...  ..   ...     ...     ...   
282622        ZOLOTIC Anastasija   F  19.0    1.80     NaN   
282623              ZOU Jingyuan   M  23.0    1.58     NaN   
282624              ZOU Jingyuan   M  23.0    1.58     NaN   
282625          ZUBIMENDI Martin   M  22.0    1.80     NaN   
282626          ZVEREV Alexander   M  24.0    1.98     NaN   

                              Team  NOC        Games  Year  Season       City  \
0                            China  CHN  1992 Summer  1992  Summer  Barcelona   
1                            China  CHN  2012 Summer  2012  Summer     London   
2                          Denmark  DEN  1920 Summer  1920  Summer  Antwerpen   
3                   Denmark/Sweden  DEN  1900 Summer  1900  Summer      Paris   
4                      Netherlands  NED  1988 Winter  1988  Winter    Calgary   
...                            ...  ...          ...   ...     ...        ...   
282622    United States of America  USA   Tokyo 2020  2020  Summer      Tokyo   
282623  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282624  People's Republic of China  CHN   Tokyo 2020  2020  Summer      Tokyo   
282625                       Spain  ESP   Tokyo 2020  2020  Summer      Tokyo   
282626                     Germany  GER   Tokyo 2020  2020  Summer      Tokyo   

                      Sport                             Event         Medal  \
0                Basketball       Basketball Men's Basketball  Pas_medaille   
1                      Judo      Judo Men's Extra-Lightweight  Pas_medaille   
2                  Football           Football Men's Football  Pas_medaille   
3                Tug-Of-War       Tug-Of-War Men's Tug-Of-War          Gold   
4             Speed Skating  Speed Skating Women's 500 metres  Pas_medaille   
...                     ...                               ...           ...   
282622            Taekwondo                       Women -57kg          Gold   
282623  Artistic Gymnastics                        Men's Team        Bronze   
282624  Artistic Gymnastics               Men's Parallel Bars          Gold   
282625             Football                               Men        Silver   
282626               Tennis                     Men's Singles          Gold   

          Country_x notes      Country_y    Population  PIB/habitant  \
0             China   NaN          China  1.371220e+09   8027.683810   
1             China   NaN          China  1.371220e+09   8027.683810   
2           Denmark   NaN        Denmark  5.676002e+06  51989.293471   
3           Denmark   NaN        Denmark  5.676002e+06  51989.293471   
4       Netherlands   NaN    Netherlands  1.693652e+07  44299.768085   
...             ...   ...            ...           ...           ...   
282622          USA   NaN  United States  3.214188e+08  56115.718426   
282623        China   NaN          China  1.371220e+09   8027.683810   
282624        China   NaN          China  1.371220e+09   8027.683810   
282625        Spain   NaN          Spain  4.641827e+07  25831.582305   
282626      Germany   NaN        Germany  8.141314e+07  41313.313995   

        Score_medaille  
0                    0  
1                    0  
2                    0  
3                    3  
4                    0  
...                ...  
282622               3  
282623               1  
282624               3  
282625               2  
282626               3  

[282627 rows x 20 columns]

In [27] used 0.5 MiB RAM in 0.14s (system mean cpu 15%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 230.9 MiB


In [28]:
#Marche aussi 
''' 
medaille_categories=['Pas_medaille','Bronze','Silver','Gold']
encode_medaille=preprocessing.OrdinalEncoder(categories=[medaille_categories])

df_complete['enc_medal']=encode_medaille.fit_transform(df_complete[['Medal']])

df_complete
''' 

" \nmedaille_categories=['Pas_medaille','Bronze','Silver','Gold']\nencode_medaille=preprocessing.OrdinalEncoder(categories=[medaille_categories])\n\ndf_complete['enc_medal']=encode_medaille.fit_transform(df_complete[['Medal']])\n\ndf_complete\n"

In [28] used 0.1 MiB RAM in 0.11s (system mean cpu 14%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 231.0 MiB


## Analyse valeurs nulles 

In [29]:
df_complete.isna().sum()

Name                   0
Sex                  156
Age                 9628
Height             67142
Weight             74386
Team                   0
NOC                    0
Games                  0
Year                   0
Season                 0
City                   0
Sport                156
Event               9278
Medal                  0
Country_x            419
notes             277510
Country_y          26221
Population         27593
PIB/habitant       35919
Score_medaille         0
dtype: int64

In [29] used 3.7 MiB RAM in 0.24s (system mean cpu 4%, single max cpu 83%), peaked 0.0 MiB above final usage, current RAM usage now 234.7 MiB


156 sports où on sait pas ce que c'est .... et des country (26 221) qui ne sont pas attribués non plus 

Pour les Country , on va merge 

In [30]:
#df_complete_test=df_complete.copy()

df_complete=pd.merge(df_complete,df_regions,on='NOC',how='left')

df_complete['Country']=df_complete['Country_y'].fillna(df_complete['Country_x'])

In [30] used 53.9 MiB RAM in 0.18s (system mean cpu 4%, single max cpu 64%), peaked 0.0 MiB above final usage, current RAM usage now 288.6 MiB


In [31]:
df_complete= df_complete.drop(['Country_x', 'Country_y'], axis=1)

In [31] used 16.4 MiB RAM in 0.16s (system mean cpu 5%, single max cpu 100%), peaked 5.2 MiB above final usage, current RAM usage now 305.0 MiB


In [32]:
liste_pays=df_regions['Country'].tolist()
liste_NOC=df_regions['NOC'].tolist()

dictio_NOC_pays=dict(zip(liste_NOC,liste_pays))

dictio_NOC_pays['SGP']='Singapour'
dictio_NOC_pays['LBN']='Liban'


df_complete['Country']=df_complete['Country'].fillna(df_complete['NOC'].map(dictio_NOC_pays))
#liste_region

#df_complete_test.isna().sum()

In [32] used -1.2 MiB RAM in 0.15s (system mean cpu 5%, single max cpu 50%), peaked 0.0 MiB above final usage, current RAM usage now 303.8 MiB


In [33]:
#test=df_complete_test[df_complete_test['Country'].isna()]

#test['Country']=test['Country'].fillna(test['NOC'].map(dictio_NOC_pays))

#test.groupby('NOC')['Name'].count().reset_index()

In [33] used 0.0 MiB RAM in 0.11s (system mean cpu 10%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 303.8 MiB


Il y a quatre NOC qui ne correspondent à aucun pays dans notre EOR : réfugiés , ROT réfugiés aussi , LBN c'est le Liban (Lebanon en anglais )

In [34]:
df_complete=df_complete.query(" NOC !='ROT' and NOC !='EOR' ")



In [34] used 12.0 MiB RAM in 0.19s (system mean cpu 10%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 315.8 MiB


In [35]:
df_complete=df_complete.drop(['notes_x','notes_y'],axis=1)

In [35] used 4.4 MiB RAM in 0.15s (system mean cpu 10%, single max cpu 67%), peaked 0.0 MiB above final usage, current RAM usage now 320.2 MiB


## Opérations sur les sports 

In [36]:
liste_sports=df_complete['Sport'].unique().tolist()

print(liste_sports)

['Basketball', 'Judo', 'Football', 'Tug-Of-War', 'Speed Skating', 'Cross Country Skiing', 'Athletics', 'Ice Hockey', 'Swimming', 'Badminton', 'Sailing', 'Biathlon', 'Gymnastics', 'Art Competitions', 'Alpine Skiing', 'Handball', 'Weightlifting', 'Wrestling', 'Luge', 'Water Polo', 'Hockey', 'Rowing', 'Bobsleigh', 'Fencing', 'Equestrianism', 'Shooting', 'Boxing', 'Taekwondo', 'Cycling', 'Diving', 'Canoeing', 'Tennis', 'Modern Pentathlon', 'Figure Skating', 'Golf', 'Softball', 'Archery', 'Volleyball', 'Synchronized Swimming', 'Table Tennis', 'Nordic Combined', 'Baseball', 'Rhythmic Gymnastics', 'Freestyle Skiing', 'Rugby Sevens', 'Trampolining', 'Beach Volleyball', 'Triathlon', 'Ski Jumping', 'Curling', 'Snowboarding', 'Rugby', 'Short Track Speed Skating', 'Skeleton', 'Lacrosse', 'Polo', 'Cricket', 'Racquets', 'Motorboating', 'Military Ski Patrol', 'Croquet', 'Jeu De Paume', 'Roque', 'Alpinism', 'Basque Pelota', 'Aeronautics', 'Cycling Road', 'Artistic Gymnastics', 'Karate', 'Trampoline Gy

In [37]:
table_sport=df_complete.groupby('Sport')['Name'].count().reset_index()

table_sport

Sport  Name
0   3x3 Basketball    56
1      Aeronautics     1
2    Alpine Skiing  8829
3         Alpinism    25
4          Archery  2461
..             ...   ...
79      Tug-Of-War   170
80      Volleyball  3666
81      Water Polo  4119
82   Weightlifting  4128
83       Wrestling  7425

[84 rows x 2 columns]

In [37] used 0.1 MiB RAM in 0.17s (system mean cpu 41%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 320.1 MiB


In [38]:
df_complete=df_complete[df_complete['Year']>=2000]

In [38] used -3.1 MiB RAM in 0.13s (system mean cpu 31%, single max cpu 100%), peaked 3.1 MiB above final usage, current RAM usage now 317.0 MiB


In [39]:
df_complete

Name Sex   Age  Height  Weight  \
1                       A Lamusi   M  23.0  170.00    60.0   
28              Antti Sami Aalto   M  26.0  186.00    96.0   
31             Jyri Tapani Aalto   M  31.0  172.00    70.0   
33            Minna Maarit Aalto   F  34.0  159.00    55.5   
40      Juhamatti Tapio Aaltonen   M  28.0  184.00    85.0   
...                          ...  ..   ...     ...     ...   
282622        ZOLOTIC Anastasija   F  19.0    1.80     NaN   
282623              ZOU Jingyuan   M  23.0    1.58     NaN   
282624              ZOU Jingyuan   M  23.0    1.58     NaN   
282625          ZUBIMENDI Martin   M  22.0    1.80     NaN   
282626          ZVEREV Alexander   M  24.0    1.98     NaN   

                              Team  NOC        Games  Year  Season  \
1                            China  CHN  2012 Summer  2012  Summer   
28                         Finland  FIN  2002 Winter  2002  Winter   
31                         Finland  FIN  2000 Summer  2000  Summer   
33                         Finland  FIN  2000 Summer  2000  Summer   
40                         Finland  FIN  2014 Winter  2014  Winter   
...                            ...  ...          ...   ...     ...   
282622    United States of America  USA   Tokyo 2020  2020  Summer   
282623  People's Republic of China  CHN   Tokyo 2020  2020  Summer   
282624  People's Republic of China  CHN   Tokyo 2020  2020  Summer   
282625                       Spain  ESP   Tokyo 2020  2020  Summer   
282626                     Germany  GER   Tokyo 2020  2020  Summer   

                  City                Sport                         Event  \
1               London                 Judo  Judo Men's Extra-Lightweight   
28      Salt Lake City           Ice Hockey   Ice Hockey Men's Ice Hockey   
31              Sydney            Badminton       Badminton Men's Singles   
33              Sydney              Sailing    Sailing Women's Windsurfer   
40               Sochi           Ice Hockey   Ice Hockey Men's Ice Hockey   
...                ...                  ...                           ...   
282622           Tokyo            Taekwondo                   Women -57kg   
282623           Tokyo  Artistic Gymnastics                    Men's Team   
282624           Tokyo  Artistic Gymnastics           Men's Parallel Bars   
282625           Tokyo             Football                           Men   
282626           Tokyo               Tennis                 Men's Singles   

               Medal    Population  PIB/habitant  Score_medaille  \
1       Pas_medaille  1.371220e+09   8027.683810               0   
28      Pas_medaille  5.482013e+06  42311.036231               0   
31      Pas_medaille  5.482013e+06  42311.036231               0   
33      Pas_medaille  5.482013e+06  42311.036231               0   
40            Bronze  5.482013e+06  42311.036231               1   
...              ...           ...           ...             ...   
282622          Gold  3.214188e+08  56115.718426               3   
282623        Bronze  1.371220e+09   8027.683810               1   
282624          Gold  1.371220e+09   8027.683810               3   
282625        Silver  4.641827e+07  25831.582305               2   
282626          Gold  8.141314e+07  41313.313995               3   

              Country  
1               China  
28            Finland  
31            Finland  
33            Finland  
40            Finland  
...               ...  
282622  United States  
282623          China  
282624          China  
282625          Spain  
282626        Germany  

[96730 rows x 18 columns]

In [39] used 0.1 MiB RAM in 0.13s (system mean cpu 24%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 317.1 MiB


## Opérations sur les Ages

In [40]:
table_sport_age=df_complete.groupby('Sport')['Age'].mean().apply(m.ceil).reset_index()

table_sport_age

Sport  Age
0        3x3 Basketball   30
1         Alpine Skiing   25
2               Archery   27
3   Artistic Gymnastics   25
4     Artistic Swimming   24
..                  ...  ...
64            Triathlon   29
65           Volleyball   27
66           Water Polo   27
67        Weightlifting   25
68            Wrestling   27

[69 rows x 2 columns]

In [40] used 2.3 MiB RAM in 0.13s (system mean cpu 22%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 319.5 MiB


In [41]:
liste_sport=table_sport_age['Sport'].tolist()
liste_age=table_sport_age['Age'].tolist()

dictio_age_sport=dict(zip(liste_sport,liste_age))

In [41] used 0.0 MiB RAM in 0.11s (system mean cpu 27%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 319.5 MiB


In [42]:
df_complete['Age']=df_complete['Age'].fillna(df_complete['Sport'].map(dictio_age_sport))

df_complete.isna().sum()

Name                  0
Sex                 156
Age                 156
Height             7658
Weight            12694
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport               156
Event              9249
Medal                 0
Population         2855
PIB/habitant       5741
Score_medaille        0
Country               0
dtype: int64

In [42] used -1.5 MiB RAM in 0.16s (system mean cpu 26%, single max cpu 100%), peaked 1.5 MiB above final usage, current RAM usage now 318.0 MiB


In [43]:
df_complete[df_complete['Age'].isna()].groupby('Medal')['Name'].count().reset_index()

Medal  Name
0  Pas_medaille   156

In [43] used -0.0 MiB RAM in 0.11s (system mean cpu 12%, single max cpu 60%), peaked 0.0 MiB above final usage, current RAM usage now 318.0 MiB


In [44]:
#df_tokyo=pd.read_csv('data/athlete_events_tokyo_cleaned.csv')

#df_tokyo.isna().sum()

In [44] used 0.0 MiB RAM in 0.10s (system mean cpu 12%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 318.0 MiB


Enfaite il y a 156 personnes qui manquent prcq c'est les qu'on a pas les sport 

In [45]:
df_complete=df_complete.dropna(subset='Age')

In [45] used 9.9 MiB RAM in 0.13s (system mean cpu 12%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 327.9 MiB


In [46]:
df_complete.isna().sum()

Name                  0
Sex                   0
Age                   0
Height             7502
Weight            12538
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport                 0
Event              9093
Medal                 0
Population         2846
PIB/habitant       5731
Score_medaille        0
Country               0
dtype: int64

In [46] used 1.0 MiB RAM in 0.15s (system mean cpu 20%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 328.9 MiB


## Opérations sur les poids et les tailles 

In [47]:
df_complete.isna().sum()

Name                  0
Sex                   0
Age                   0
Height             7502
Weight            12538
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport                 0
Event              9093
Medal                 0
Population         2846
PIB/habitant       5731
Score_medaille        0
Country               0
dtype: int64

In [47] used -0.4 MiB RAM in 0.15s (system mean cpu 27%, single max cpu 100%), peaked 0.4 MiB above final usage, current RAM usage now 328.5 MiB


In [48]:
table_taille=df_complete[df_complete['Height'].isna()==False].groupby(['Sport'])['Height'].mean().apply(m.ceil).reset_index()

liste_sport=table_taille['Sport'].tolist()
#liste_sex=table_taille['Sex'].tolist()
liste_taille=table_taille['Height'].tolist()
print(table_taille)
liste_sport_sex=[]
for i in range(len(liste_sport)):
    liste_sport_sex.append((liste_sport[i],liste_taille[i]))
#print(liste_sport_sex)

dictio_sport_sex=dict(zip(liste_sport,liste_taille))



                  Sport  Height
0        3x3 Basketball       2
1         Alpine Skiing     176
2               Archery     174
3   Artistic Gymnastics       2
4     Artistic Swimming       2
..                  ...     ...
62            Triathlon     169
63           Volleyball     161
64           Water Polo     169
65        Weightlifting     159
66            Wrestling     171

[67 rows x 2 columns]
In [48] used 5.8 MiB RAM in 0.13s (system mean cpu 25%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 334.3 MiB


In [49]:
df_complete['Height'] = df_complete['Height'].fillna(df_complete['Sport'].map(dictio_sport_sex))


df_complete.isna().sum()

<ipython-input-49-30982cdfd39c>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_complete['Height'] = df_complete['Height'].fillna(df_complete['Sport'].map(dictio_sport_sex))


Name                  0
Sex                   0
Age                   0
Height              159
Weight            12538
Team                  0
NOC                   0
Games                 0
Year                  0
Season                0
City                  0
Sport                 0
Event              9093
Medal                 0
Population         2846
PIB/habitant       5731
Score_medaille        0
Country               0
dtype: int64

In [49] used 0.2 MiB RAM in 0.17s (system mean cpu 17%, single max cpu 75%), peaked 0.0 MiB above final usage, current RAM usage now 334.6 MiB


In [50]:
#df_complete[df_complete['Height'].isna()]

In [50] used -0.0 MiB RAM in 0.11s (system mean cpu 15%, single max cpu 67%), peaked 0.0 MiB above final usage, current RAM usage now 334.6 MiB


In [51]:
table_poids=df_complete[df_complete['Weight'].isna()==False].groupby(['Sport'])['Weight'].mean().apply(m.ceil).reset_index()

liste_sport_2=table_poids['Sport'].tolist()
#liste_sex=table_taille['Sex'].tolist()
liste_poids=table_poids['Weight'].tolist()



dictio_sport_poids=dict(zip(liste_sport_2,liste_poids))

In [51] used 3.8 MiB RAM in 0.15s (system mean cpu 29%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 338.4 MiB


In [52]:
df_complete['Weight'] = df_complete['Weight'].fillna(df_complete['Sport'].map(dictio_sport_poids))


df_complete.info()

<class 'pandas.core.frame.DataFrame'>
Index: 96574 entries, 1 to 282626
Data columns (total 18 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Name            96574 non-null  object 
 1   Sex             96574 non-null  object 
 2   Age             96574 non-null  float64
 3   Height          96415 non-null  float64
 4   Weight          94584 non-null  float64
 5   Team            96574 non-null  object 
 6   NOC             96574 non-null  object 
 7   Games           96574 non-null  object 
 8   Year            96574 non-null  int64  
 9   Season          96574 non-null  object 
 10  City            96574 non-null  object 
 11  Sport           96574 non-null  object 
 12  Event           87481 non-null  object 
 13  Medal           96574 non-null  object 
 14  Population      93728 non-null  float64
 15  PIB/habitant    90843 non-null  float64
 16  Score_medaille  96574 non-null  int64  
 17  Country         96574 non-null  obj

<ipython-input-52-f91662992dfc>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_complete['Weight'] = df_complete['Weight'].fillna(df_complete['Sport'].map(dictio_sport_poids))


In [53]:
df_complete=df_complete[df_complete['Season']=='Summer']

df_complete

Name Sex   Age  Height  Weight  \
1                        A Lamusi   M  23.0  170.00    60.0   
31              Jyri Tapani Aalto   M  31.0  172.00    70.0   
33             Minna Maarit Aalto   F  34.0  159.00    55.5   
57           Timo Antero Aaltonen   M  31.0  189.00   130.0   
79      Ragnhild Margrethe Aamodt   F  27.0  163.00    82.0   
...                           ...  ..   ...     ...     ...   
282622         ZOLOTIC Anastasija   F  19.0    1.80    69.0   
282623               ZOU Jingyuan   M  23.0    1.58     NaN   
282624               ZOU Jingyuan   M  23.0    1.58     NaN   
282625           ZUBIMENDI Martin   M  22.0    1.80    70.0   
282626           ZVEREV Alexander   M  24.0    1.98    72.0   

                              Team  NOC        Games  Year  Season     City  \
1                            China  CHN  2012 Summer  2012  Summer   London   
31                         Finland  FIN  2000 Summer  2000  Summer   Sydney   
33                         Finland  FIN  2000 Summer  2000  Summer   Sydney   
57                         Finland  FIN  2000 Summer  2000  Summer   Sydney   
79                          Norway  NOR  2008 Summer  2008  Summer  Beijing   
...                            ...  ...          ...   ...     ...      ...   
282622    United States of America  USA   Tokyo 2020  2020  Summer    Tokyo   
282623  People's Republic of China  CHN   Tokyo 2020  2020  Summer    Tokyo   
282624  People's Republic of China  CHN   Tokyo 2020  2020  Summer    Tokyo   
282625                       Spain  ESP   Tokyo 2020  2020  Summer    Tokyo   
282626                     Germany  GER   Tokyo 2020  2020  Summer    Tokyo   

                      Sport                         Event         Medal  \
1                      Judo  Judo Men's Extra-Lightweight  Pas_medaille   
31                Badminton       Badminton Men's Singles  Pas_medaille   
33                  Sailing    Sailing Women's Windsurfer  Pas_medaille   
57                Athletics      Athletics Men's Shot Put  Pas_medaille   
79                 Handball     Handball Women's Handball          Gold   
...                     ...                           ...           ...   
282622            Taekwondo                   Women -57kg          Gold   
282623  Artistic Gymnastics                    Men's Team        Bronze   
282624  Artistic Gymnastics           Men's Parallel Bars          Gold   
282625             Football                           Men        Silver   
282626               Tennis                 Men's Singles          Gold   

          Population  PIB/habitant  Score_medaille        Country  
1       1.371220e+09   8027.683810               0          China  
31      5.482013e+06  42311.036231               0        Finland  
33      5.482013e+06  42311.036231               0        Finland  
57      5.482013e+06  42311.036231               0        Finland  
79      5.195921e+06  74400.369777               3         Norway  
...              ...           ...             ...            ...  
282622  3.214188e+08  56115.718426               3  United States  
282623  1.371220e+09   8027.683810               1          China  
282624  1.371220e+09   8027.683810               3          China  
282625  4.641827e+07  25831.582305               2          Spain  
282626  8.141314e+07  41313.313995               3        Germany  

[78790 rows x 18 columns]

In [53] used 1.0 MiB RAM in 0.14s (system mean cpu 22%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 340.5 MiB


## Moyenne par pays pour les sports qu'on ne connait pas 


In [54]:
table_pays_1=df_complete[df_complete['Weight'].isna()==False].groupby(['Country'])['Weight'].mean().apply(m.ceil).reset_index()

liste_pays_1=table_pays_1['Country'].tolist()
#liste_sex=table_taille['Sex'].tolist()
liste_poids_1=table_pays_1['Weight'].tolist()



dictio_pays_poids_1=dict(zip(liste_pays_1,liste_poids_1))
df_complete['Weight'] = df_complete['Weight'].fillna(df_complete['Country'].map(dictio_pays_poids_1))

In [54] used 0.4 MiB RAM in 0.13s (system mean cpu 15%, single max cpu 67%), peaked 0.0 MiB above final usage, current RAM usage now 340.9 MiB


In [55]:
table_pays_2=df_complete[df_complete['Height'].isna()==False].groupby(['Country'])['Height'].mean().apply(m.ceil).reset_index()

liste_pays_2=table_pays_2['Country'].tolist()
#liste_sex=table_taille['Sex'].tolist()
liste_poids_2=table_pays_2['Height'].tolist()



dictio_pays_poids_2=dict(zip(liste_pays_2,liste_poids_2))
df_complete['Height'] = df_complete['Height'].fillna(df_complete['Country'].map(dictio_pays_poids_2))

#df_complete.isna().sum()

In [55] used 6.4 MiB RAM in 0.14s (system mean cpu 9%, single max cpu 75%), peaked 0.0 MiB above final usage, current RAM usage now 347.2 MiB


In [56]:
#df_complete.to_csv('data/df_merge_FLO.csv', index=False)

In [56] used 0.0 MiB RAM in 0.11s (system mean cpu 17%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 347.2 MiB


In [57]:
from bs4 import BeautifulSoup
import requests

In [57] used 12.9 MiB RAM in 0.31s (system mean cpu 18%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 360.2 MiB


In [58]:
url='https://www.topendsports.com/events/summer/hosts/list.htm'
response=requests.get(url)

In [58] used -6.7 MiB RAM in 1.03s (system mean cpu 16%, single max cpu 100%), peaked 7.8 MiB above final usage, current RAM usage now 353.5 MiB


In [59]:
soup=BeautifulSoup(response.text,'lxml')

In [59] used -0.0 MiB RAM in 0.12s (system mean cpu 14%, single max cpu 100%), peaked 0.0 MiB above final usage, current RAM usage now 353.5 MiB


In [62]:
year_elements=soup.findAll('th',class_='Olympiad')
L_pays=[]
for pays in L_pays:
    L_pays.append(pays.text)
    
L_pays

[]

In [62] used -65.2 MiB RAM in 0.11s (system mean cpu 20%, single max cpu 30%), peaked 0.0 MiB above final usage, current RAM usage now 142.1 MiB


In [ ]:
years = []
for year_element in year_elements:
    year_text = year_element.find_next_sibling('td').text
    year = int(year_text)
    years.append(year)

years